### PR TITLE
prevent double snap

### PR DESCRIPTION
**Pull request by reco**
add new setting prevent-double-snap. is set to true prevent to do 2 snaps with a short snap-duration.

### CODE REVIEW

**Comment by reco** [comment]
you can see an example here... https://moniquepean.imago.io

**Comment by joelmukuthu** [comment]
This is really nice, thanks! Since this is a fix for a known bug I think we can adopt this as the default behaviour so no need for the new attribute, what do you think? 

`dist/snapscroll.js` is a generated file, so you'll have to move that code to `src/directives/snapscroll.js` so that it doesn't get overwritten the next time I'm releasing a new version. I'm also interested in seeing what effect this change will have on the tests. If [you'll allow me to](https://help.github.com/articles/allowing-changes-to-a-pull-request-branch-created-from-a-fork/) I can take care of that :)

File: dist/angular-snapscroll.js [hunk 1]
```diff
@@ -453,12 +473,20 @@
                     function bindWheel() {
                         wheelie.bind(element, {
                             up: function (e) {
+                                if (Math.abs(e.wheelDelta) > scope.oldWheelDelta ) {
```
Comment (joelmukuthu) [inline dist/angular-snapscroll.js:476]: Shouldn't this be `Math.abs(e.wheelDelta) < scope.oldWheelDelta`?

**Comment by joelmukuthu** [comment]
I'm also wondering if this fix qualifies to be in https://github.com/joelmukuthu/angular-wheelie..

File: dist/angular-snapscroll.js (hunk 1 above)
Comment (reco) [reply dist/angular-snapscroll.js:476]: i thought that too before. but actually we should allow another snap when another scroll is done which increases the delta.

**Comment by reco** [comment]
i would put it into the snapscroll. wheelie does in my opinion what it needs to do. leave this up to you.

ok i will remove the new attribute add it to the src. sure feel free to merge it test sure.. all yours

**Comment by reco** [comment]
hey joelmukuthu i modified the src file. i cannot get the tests to pass though.

**Comment by joelmukuthu** [comment]
i had a strong suspicion that that would happen. i'll look into that

**Comment by joelmukuthu** [comment]
Hey @reco, sorry for the silence. I broke up `preventSnap` into `preventUp` and `preventDown` and simplified the toggling. PTAL and let me know if it still does the same thing as you intended. Btw that change got the failing tests to go down to from 16 to 9 :)

**Comment by joelmukuthu** [comment]
@reco, this is now ready to be reviewed/merged. could you test the new built file in `dist/angular-snapscroll.min.js` against your site to see if it works as expected?

**Comment by joelmukuthu** [comment]
Rolling with it :)

**Pull request merged**